lib/improver/nbhood/square_kernel.py

```diff
@@ -108,27 +108,17 @@ def cumulate_array(cube):
                 y dimensions, so will generally be a slice of a cube.
 
         Returns:
-            (tuple) : tuple containing:
-                **cube** (Iris.cube.Cube):
-                    Cube to which the cumulative summing along the y and x
-                    direction has been applied.
-                **nan_masks** (list):
-                    List of numpy arrays to be used to set the values within
-                    the data of the output cube to be NaN.
+            summed_cube (Iris.cube.Cube):
+                Cube to which the cumulative summing
+                along the y and x direction has been applied.
         """
-        yname = cube.coord(axis="y").name()
-        xname = cube.coord(axis="x").name()
-        cubelist = iris.cube.CubeList([])
-        nan_masks = []
+        summed_cube = cube.copy()
         data = cube.data.astype(np.longdouble)
-        nan_mask = np.isnan(data)
-        data[nan_mask] = 0
         data_summed_along_y = np.cumsum(data, axis=0)
```

Comment:
Does this assume which way round the y and x coordinates are in the array? If so we should probably note it in the docstring.

Reply:
Because we slice the cube over y_axis and x_axis this should always be axis 0 but I can add a note to this affect.